### ISSUE TITLE
`Unexpected store used for installation` not case-insensitive                                  

### ISSUE THREAD

**Issue by nickpape-msft** [pnpm/pnpm#996]
### Code to reproduce the issue:

Run the following commands:
```
pnpm install --no-optional --store d:\code\web-build-tools\common\temp\pnpm-store --no-lock
pnpm install --no-optional --store D:\code\web-build-tools\common\temp\pnpm-store --no-lock
```

We are spinning up the PNPM process and calculating the store path by building up a path that is ultimately derived from `process.cwd()`. However, the core Node API does not normalize the drive letter. Nor does any of the [path methods](https://github.com/nodejs/node-v0.x-archive/pull/8385).

We could normalize the drive letter in Rush, but it seems like it would be better to make the fix in PNPM.

**Proposed fix:**
Consider doing `path.relative()` instead of a string compare.

### Expected behavior:

Everything is OK since the two paths are effectively the same.

### Actual behavior:

```
ERROR  Unexpected store used for installation                                                         
                                                                                                       
expected: d:\code\web-build-tools\common\temp\pnpm-store\2                                             
actual: D:\code\web-build-tools\common\temp\pnpm-store\2                                               
                                                                                                       
If you want to use the new store, run the same command with the --force parameter.                     
```

**Comment by nickpape-msft** [comment]
Related PR: https://github.com/pnpm/supi/pull/39/files

**Comment by zkochan** [comment]
:ship: published with `pnpm@1.30.0`
